src/cluster_legacy.c

```diff
@@ -4928,6 +4928,11 @@ int clusterDelSlot(int slot) {
 
     if (!n) return C_ERR;
 
+    /* Cleanup the empty dicts if myself is the owner. */
+    if (n == myself) {
+        kvstoreFreeDictIfEmpty(server.db->keys, slot);
+        kvstoreFreeDictIfEmpty(server.db->expires, slot);
```

Comment:
let's handle sharded pubsub too.

Reply:
I thought the removeChannelsInSlot below would handle it (KVSTORE_FREE_EMPTY_DICTS), but after looking it again, it didn’t conflict, i added it

Reply:
ohh, you're right. it's not needed..
but well, you're also right that it doesn't cause any harm, so let's keep it.